e2e/libs/recurringjob/rest.py

```diff
@@ -45,139 +47,137 @@ def add_to_volume(self, job_name, volume_name):
+                jobs = []
+                groups = []
+                for item in list:
+                    if item['isGroup']:
+                        groups.append(item['name'])
+                    else:
+                        jobs.append(item['name'])
+                return jobs, groups
+            except Exception as e:
+                logging(f"Getting volume {volume} recurringjob list error: {e}")
+                time.sleep(self.retry_interval)
 
     def _wait_for_cron_job_create(self, job_name):
         created = False
-        for _ in range(RETRY_COUNTS):
+        for _ in range(self.retry_count):
             job = self.batch_v1_api.list_namespaced_cron_job(
                 'longhorn-system',
                 label_selector=f"recurring-job.longhorn.io={job_name}")
             if len(job.items) != 0:
                 created = True
                 break
-            time.sleep(RETRY_INTERVAL)
+            time.sleep(self.retry_interval)
         assert created
 
     def _wait_for_cron_job_delete(self, job_name):
         deleted = False
-        for _ in range(RETRY_COUNTS):
+        for _ in range(self.retry_count):
             job = self.batch_v1_api.list_namespaced_cron_job(
                 'longhorn-system',
                 label_selector=f"recurring-job.longhorn.io={job_name}")
             if len(job.items) == 0:
                 deleted = True
                 break
-            time.sleep(RETRY_INTERVAL)
+            time.sleep(self.retry_interval)
         assert deleted
 
     def check_jobs_work(self, volume_name):
-        # check if snapshot/backup is really created by the
-        # recurringjob following the cron schedule
-        # currently only support checking normal snapshot/backup
-        # every 1 min recurringjob
         jobs, _ = self.get_volume_recurringjobs_and_groups(volume_name)
         for job_name in jobs:
             job = self.get(job_name)
             logging(f"Checking recurringjob {job}")
-            if job['task'] == 'snapshot' and job['cron'] == '* * * * *':
-                period_in_sec = 60
-                self._check_snapshot_created_in_time(volume_name, job_name, period_in_sec)
-            elif job['task'] == 'backup' and job['cron'] == '* * * * *':
-                period_in_sec = 60
-                self._check_backup_created_in_time(volume_name, period_in_sec)
-
-    def _check_snapshot_created_in_time(self, volume_name, job_name, period_in_sec):
+            if job['task'] == 'snapshot':
+                self._check_snapshot_created(volume_name, job_name)
+            elif job['task'] == 'backup':
+                self._check_backup_created(volume_name)
+
+    def _check_snapshot_created(self, volume_name, job_name):
         # check snapshot can be created by the recurringjob
         current_time = datetime.utcnow()
         current_timestamp = current_time.timestamp()
-
+        logging(f"Recorded current time = {current_time}, timestamp = {current_timestamp}")
```

Comment:
Could you explain the purpose of this log?

Reply:
Removed. Thank you!